src/app/content/components/ContentLink.tsx

```diff
@@ -54,6 +55,9 @@ export const ContentLink = (props: React.PropsWithChildren<Props>) => {
     ...anchorProps
   } = props;
   const {url, params} = getBookPageUrlAndParams(book, page);
+  const navigationMatch = assertDefined(
+    createNavigationMatch(page.id, book, params),
+    `Page ${page.id} does not exist in book ${book.id}`);
```

Comment:
why might this not return anything? would it help to pass the whole page instead of just the id? we weren't previously checking if the given page was in the given book.